flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/TableSinkITCase.scala

```diff
@@ -255,7 +253,7 @@ class TableSinkITCase(mode: StateBackendMode) extends StreamingWithStateTestBase
       "+I[jason, 1, null, null, null, null]",
       "+I[jason, 1, null, null, null, null]"
     )
-    assertEquals(expected.sorted, result.sorted)
+    assertThat(result.sorted).isEqualTo(expected.sorted)
```

Comment:
Could we remove junit4 dependency in table-planner to make sure we have migrated all tests ?

Reply:
Currently, it is very difficult to remove JUnit4 because its dependency is imported from `flink-parent`. To remove the dependency, it may be necessary to declare JUnit4 as `optional` in `flink-parent` and then add JUnit4 dependency to the submodules as needed.

To avoid adding JUnit 4 tests in the `flink-table-planner` module, I'm wondering if we can add `ArchUnit` rules as a transitional solution.

Reply:
Your detailed explanation makes sense to me. +1